### PR TITLE
Do not dissable all repos every-time

### LINKED ISSUE #148: Disabling all repositories in RHEL Dockerfiles is problematic
The cases where we need to enable different set of repositories that what we expect -- then disabling all repos is not possible. A usual trick to build RHEL dockerfiles on an unsubscribed machine is to add a different repo in the beginning of the Dockerfile. Since RHEL Dockerfiles currently disable all repos and enable just specific ones, it means the additional repo file won't be used.

A solution for this might be either of those (or both):
* disable only repos that are coming from subscription (those matching `rhel-*`)
* disable repos only when on subscribed machine (is a simple check like `[ -f /etc/pki/entitlement/*.pem ]` good enough?)

If we use both, it could look like:
```
RUN [ -f /etc/pki/entitlement/*.pem ] && yum-config-manager --disable rhel-*
```

### PR DESCRIPTION
**Pull request by hhorak**
It causes troubles when some other repositories are added by user or build system.
More discussion happens in #148.
Fixes #148.

### CODE REVIEW

**Comment by pkubatrh** [comment]
rhel7 test run fails because of

```
Installed: libselinux-2.5-12.el7.x86_64 (@rhel-7-server-htb-rpms)
```

**Comment by hhorak** [comment]
It looks like this check does not work as I thought: `test -f '/etc/pki/entitlement/*.pem'`

**Comment by pkubatrh** [comment]
>It looks like this check does not work as I thought: test -f '/etc/pki/entitlement/*.pem'

That is likely, since `test` only takes a single argument. How about using this instead?

```
ls /etc/pki/entitlement/*.pem 2>/dev/null
```

**Comment by hhorak** [comment]
`ls` should work as well, although "the internet" suggests using `for`, which might work better in some corner cases, since it does not require stderr&stdout redirection to `/dev/null`.

**Comment by pkubatrh** [comment]
[test]

**Comment by hhorak** [comment]
We can have the list of repos in the script, since it is same in all containers (almost). Then, when beta needs to be used, we might set some argument for the build (taken from the builder environment)..

**Comment by hhorak** [comment]
> We can have the list of repos in the script, since it is same in all containers (almost).

Actually it's not, I have a second thoughts about containers like devtoolset or rust/go/llvm from devtools channel and others, potentially even not from RHSCL, which would still be fine with s2i-core and s2i-base, but will definitely have different set of repositories. So, I'm actually staying with the original design and every container should specify the list of repos manually.

**Comment by pkubatrh** [comment]
How about having the list of common repositories (rhel-7-server-rpms rhel-7-server-optional-rpms) included directly inside the new script and additional repositories (rhscl, devtools) to be enabled would be added as arguments to the script?

**Comment by praiskup** [comment]
Seems like a nice idea to bake the list into the script.  Otherwise looks fine.

**Comment by hhorak** [comment]
Ok, let's have the `rhel-7-server-rpms` and `rhel-7-server-optional-rpms` be the default set. The last commit also adds two intentionally undocumented variables that might be used for changing the behaviour of the script. They may be documented once we figure out that there are some real use cases for those variables.

File: core/root/usr/bin/prepare-yum-repositories [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  yum-config-manager --disable rhel-\* &> /dev/null
+fi
+
+repos="${DEFAULT_REPOS:-}"
+if [ $# -gt 0 ] ; then
+  repos="${repos:-} $@"
+fi
+
+if [ ${SKIP_REPOS_ENABLE} = false -a -n "${repos}" ] ; then
+  yum-config-manager --enable "${repos}"
```
Comment (praiskup) [inline core/root/usr/bin/prepare-yum-repositories]: Slightly cleaner version (without the need to have `$repos` variable):
```bash
if ! ${SKIP_REPOS_ENABLE-false}; then
  yum-config-manager --enable $DEFAULT_REPOS "$@"
fi
```

File: core/root/usr/bin/prepare-yum-repositories [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# This script is used to prepare yum repositories, that are given as arguments.
+
+set -e
+
+# DEFAULT_REPOS and SKIP_REPOS_ENABLE are intentionally undocumented, but might
+# be used if we need to change this behaviour.
+# Once we realize there are real use cases for using those variables, we should
+# document them properly.
+DEFAULT_REPOS="rhel-7-server-rpms rhel-7-server-optional-rpms"
```
Comment (praiskup) [inline core/root/usr/bin/prepare-yum-repositories]: This defines unconditionally, you can use:
```
: ${DEFAULT_REPOS-"rhel-7-server-rpms rhel-7-server-optional-rpms"}
```

File: core/root/usr/bin/prepare-yum-repositories (hunk 1 above)
Comment (hhorak) [reply core/root/usr/bin/prepare-yum-repositories]: Thanks, will use this.. :)

**Comment by praiskup** [comment]
looks fine

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
lgtm as well. The test failures are because of a bug in varnish (https://github.com/sclorg/varnish-container/issues/31). Other images before varnish have been successfully tested so it is unlikely that this introduces any issues.

Merging.
